source/concepts/naming-conventions.md

```diff
@@ -1,120 +1,117 @@
+If `controller:application` is provided, Ember.js will set an
+instance of `controller:application` as the controller for the
 template. This means that the template will get its properties from
 the controller.
 
-If your app provides an `App.ApplicationRoute`, Ember.js will invoke
+If your app provides an route at `app/routes/application.js` Ember.js will invoke
 [the][1] [router's][2] [hooks][3] first, before rendering the
-`application` template. Hooks are implemented as methods and provide 
-you access points within an Ember object's lifecycle to intercept and 
-execute code to modify the default behavior at these points to meet 
+`application` template. Hooks are implemented as methods and provide
+you access points within an Ember object's lifecycle to intercept and
+execute code to modify the default behavior at these points to meet
 your needs. Ember provides several hooks for you to utilize for various
-purposes (e.g. `model`, `setupController`, etc). In the example below 
-`App.ApplicationRoute`, which is an `Ember.Route` object, implements 
+purposes (e.g. `model`, `setupController`, etc). In the example below
+`App.ApplicationRoute`, which is an `Ember.Route` object, implements
 the `setupController` hook.
 
-[1]: /guides/routing/specifying-a-routes-model
-[2]: /guides/routing/setting-up-a-controller
-[3]: /guides/routing/rendering-a-template
+[1]: ../../routing/specifying-a-routes-model
+[2]: ../../routing/setting-up-a-controller
+[3]: ../../routing/rendering-a-template
 
 Here's a simple example that uses a route, controller, and template:
 
-```javascript
-App.ApplicationRoute = Ember.Route.extend({
+```app/routes/application.js
+export default Ember.Route.extend({
   setupController: function(controller) {
     // `controller` is the instance of ApplicationController
     controller.set('title', "Hello world!");
   }
 });
+```
 
-App.ApplicationController = Ember.Controller.extend({
+```app/controllers/application.js
+export default Ember.Controller.extend({
   appName: 'My First Example'
 });
 ```
 
-```handlebars
-<!-- application template -->
+```app/templates/application.hbs
 <h1>{{appName}}</h1>
-
 <h2>{{title}}</h2>
 ```
 
-In Ember.js applications, you will always specify your controllers
+In Ember.js applications, you will always provide your objects
 as **classes**, and the framework is responsible for instantiating
-them and providing them to your templates.
+them and providing them to your templates at runtime through the resolver.
+
 
-This makes it super-simple to test your controllers, and ensures that
-your entire application shares a single instance of each controller.
 
 ## Simple Routes
 
-Each of your routes will have a controller, and a template with the 
+Each of your routes will have a controller, and a template with the
 same name as the route.
 
 Let's start with a simple router:
 
-```javascript
-App.Router.map(function() {
+```app/router.js
+export default Ember.Router.extend().map(function(){
```

Comment:
Ember CLI uses the following default `app/router.js`, should we use the default blueprint instead of what is here?

```javascript
var Router = Ember.Router.extend({
  location: config.locationType
});

Router.map(function() {
});

export default Router;
```